modules/ingest-common/src/main/java/org/elasticsearch/ingest/common/ScriptProcessor.java

```diff
@@ -80,7 +84,7 @@ public Factory(ScriptService scriptService) {
         @Override
         public ScriptProcessor create(Map<String, Processor.Factory> registry, String processorTag,
                                       Map<String, Object> config) throws Exception {
-            String lang = readStringProperty(TYPE, processorTag, config, "lang");
+            String lang = readStringProperty(TYPE, processorTag, config, "lang", "painless");
```

Comment:
Why is this hardcoded at all, and not just using the ScriptService and the default that is configured there?